src/riscv/lib/tests/test_proofs.rs

```diff
@@ -31,55 +28,73 @@ use octez_riscv::state_backend::verify_backend::Verifier;
 use octez_riscv::stepper::Stepper;
 use octez_riscv::stepper::StepperStatus;
 use octez_riscv::stepper::pvm::PvmStepper;
+use octez_riscv_test_utils::*;
 use rand::Rng;
 
-#[test]
-fn test_jstz_proofs_one_step() {
-    test_jstz_proofs::<M64M, TestCacheConfig>(false, PvmStepper::verify_proof)
-}
-
-#[test]
-fn test_jstz_proofs_one_step_stream() {
-    test_jstz_proofs::<M64M, TestCacheConfig>(false, PvmStepper::verify_proof_using_raw_bytes)
+// Run the same tests over multiple kernels
+macro_rules! proof_tests {
+    ($kernel:ident) => {
+        paste::paste! {
+            #[test]
+            fn [<test_ $kernel:lower _proofs_one_step>]() {
+                test_proofs::<M64M, TestCacheConfig>(false, PvmStepper::verify_proof, $kernel)
+            }
+
+            #[test]
+            fn [<test_ $kernel:lower _proofs_one_step_stream>]() {
+                test_proofs::<M64M, TestCacheConfig>(false, PvmStepper::verify_proof_using_raw_bytes, $kernel)
+            }
+
+            #[test]
+            #[ignore]
+            fn [<test_ $kernel:lower _proofs_full>]() {
+                test_proofs::<M64M, TestCacheConfig>(true, PvmStepper::verify_proof, $kernel
+)
+            }
+
+            #[test]
+            #[ignore]
+            fn [<test_ $kernel:lower _proofs_full_stream>]() {
+                test_proofs::<M64M, TestCacheConfig>(true, PvmStepper::verify_proof_using_raw_bytes, $kernel
+)
+            }
+
+            #[test]
+            fn [<test_ $kernel:lower _initial_proof_regression>]() {
+                test_initial_proof_regression($kernel
+)
+            }
+        }
+    };
 }
 
-#[test]
-#[ignore]
-fn test_jstz_proofs_full() {
-    test_jstz_proofs::<M64M, TestCacheConfig>(true, PvmStepper::verify_proof)
-}
-
-#[test]
-#[ignore]
-fn test_jstz_proofs_full_stream() {
-    test_jstz_proofs::<M64M, TestCacheConfig>(true, PvmStepper::verify_proof_using_raw_bytes)
-}
+proof_tests!(JSTZ);
+proof_tests!(ETHERLINK);
```

Comment:
This does not strike me as a sufficient amount of duplication to justify a macro. I wonder what else impacted the decision to go with macros.

Reply:
Just for the sake of discussion, linking my other comment about the topic [here](https://github.com/tezos/riscv-pvm/pull/370#discussion_r2397775720)

Reply:
I've removed the macro [here](https://github.com/tezos/riscv-pvm/compare/fb90774e614d282ec94dae035da7968a44fe66ce..2573629d002e996a741e3cb85476e2d9d9540b97).